### ISSUE TITLE
Render project and user avatars

### ISSUE THREAD

**Issue by LeoHuckvale** [marcells/node-build-monitor#25]
It might be useful to render project avatars (and possibly users) next to builds. This should make it easier to recognise specific projects when you have several on the screen.

This might not be available in all services (would require some research).

GitLab
-------

- [The projects API](http://docs.gitlab.com/ee/api/projects.html#get-single-project): GET /projects/:id -> JSON["avatar_url"]
- [The users API](http://docs.gitlab.com/ee/api/users.html#for-user): GET /users/:id -> JSON["avatar_url"]

**Comment by marcells** [comment]
Good idea. It's a bigger change. I will think about about a solution. Currently the monitor it self could be reached from the outside, while the build servers are only accessible in the intranet. In this situation the image urls would break. To be consistent the avatar images have to be forwarded by the monitor.

**Comment by LeoHuckvale** [comment]
Yes, I hadn't considered that - I guess you'd rather not be storing images anywhere.

Similar problem outlined [in this Stack Overflow question](http://stackoverflow.com/questions/32960113/displaying-private-s3-images-in-node-express-app) - the solution seems to be to have a route on the server to dynamically obtain the image from the private source. I suppose in this instance you'd have to implement it differently for each service, but you'd be doing that anyway...

**Comment by marcells** [comment]
Thanks. That's exactly the way, I already considered how to implement it. ;-) I think, I will add a general option to turn this feature off. Maybe the avatar pics don't match to visual design of the monitor for some users. Now I just need time to implement it. ;-)
